backend/assignments/1/src/main/java/cryptotradingplatform/Trader.java

```diff
@@ -0,0 +1,98 @@
+package cryptotradingplatform;
+
+//import org.graalvm.collections.Pair;
+import javafx.util.Pair;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Trader {
+    private String firstName;
+    private String lastName;
+    private String phone;
+
+    public long getProfit() {
+        return profit;
+    }
+
+    public void setProfit(long profit) {
+        this.profit = profit;
+    }
+
+    private long profit;
+
+
+
+    private Map<String, Pair<Long,Double>> purchasedCoins;
+
+    public void purchaseCoin(String symbol, long quantity, double price){
+        if (purchasedCoins.containsKey(symbol)){
+            purchasedCoins.put(symbol, new Pair(purchasedCoins.get(symbol).getKey() + quantity, price));
+        }
+        else purchasedCoins.put(symbol,new Pair(quantity, price));
+    }
+
+    public boolean sellCoin(String symbol, long quantity, double price){
+        if(purchasedCoins.containsKey(symbol) && purchasedCoins.get(symbol).getKey()>=quantity){
+            profit += (price - purchasedCoins.get(symbol).getValue()) * quantity;
+            purchasedCoins.put(symbol, new Pair(purchasedCoins.get(symbol).getKey() - quantity, purchasedCoins.get(symbol).getValue()));
+            return true;
+        }
+        return false;
```

Comment:
Lets not return a boolean here, rather a better way is to throw an exception and catch it when it's called.